### DISCUSSION TITLE
Middlewares enhancer

### DISCUSSION THREAD

**Discussion by ghost** [bukhalo/nestjs-telegraf#873] [Q&A]
Hello

I found, that we have only one way of using middlewares 

`stage.middleware() // register stage`

`registerUpdates() // register updates (after stage)`

But, there is no way to setup @Use, which will call before the stage. Or @StageUse - to call use on `stage`. 

I can create a PR, but i need to understand, which names of decorators to use?

**Comment by felinto-dev** [comment]
Did you already try it?

`main.ts`

```
import { Context, Telegraf } from 'telegraf';

async function bootstrap() {
  const bot: Telegraf<Context> = app.get(getBotToken());
  bot.use('CUSTOM_MIDDLEWARE_GOES_HERE...')
}
```

Source: https://nestjs-telegraf.vercel.app/getting-updates

PS: I did a test using the "session()" middleware and it worked as expected.

**Comment by ghost** [comment]
@felinto-dev Hello

Thank you for answer. I see how i can fix my problem with your way. But for this one, it will be complicated to use service methods. 

Maybe it's better to add decorators for `use`, Which will be call before the stage.middleware() or on stage

Just for now we a little bit limited
